Maskbook/Persona/PersonaImportHandler.swift

```diff
@@ -35,60 +35,107 @@ class PersonaImportPrivateKeyHandler {
     
     private var disposeBag = Set<AnyCancellable>()
     
-    func restoreFromPrivateKey(text: String) {
-        let decodedData = Data(base64URLEncoded: text)
-
-        let unpackedData = try? decodedData?.unpack() as? [String: Any]
-        guard let d = unpackedData?["d"] as? String else {
-            log.debug("private key error", source: "persona")
-            return
-        }
-        let personaRecord = PersonaRepository.queryPersonas(identifiers: nil).filter {
-            guard let privateKey = $0.privateKey else { return false }
-            guard let data = privateKey.data(using: .utf8) else { return false }
-            let json = try? JSON(data: data)
-            guard let dInPersona = json?["d"].string else { return false }
-            return d == dInPersona
-        }.first
-        
-        if let personaRecord = personaRecord {
+    func checkExistAndRestore(from item: PersonaImportItem) {
+        if let personaRecord = currentPersonaRecord(for: item) {
             if !personaRecord.hasLogout {
                 showRestoreAlreadyExistedAlert()
                 return
             }
-            let names = personaManager.personaRecordsSubject.value.map(\.nickname)
-            if let nickname = personaRecord.nickname, names.contains(nickname) {
-                let renameViewModel = RenameViewModel(
-                    originalName: "",
-                    title: L10n.Scene.Personas.Create.createPersona,
-                    dismissOnConfirm: false
-                ) { [weak self] name, viewModel in
-                    guard let self = self else { return }
-                    if names.contains(name) {
-                        self.personaNicknameDuplicated()
-                        return
-                    }
-                    viewModel.dismissSignal.send {
-                        PersonaRepository.updatePersonaNickname(identifier: personaRecord.nonOptionalIdentifier, nickname: name)
-                        self.setCurrentPersona(persona: personaRecord)
-                    }
+            renameAndSetCurrentPersona(name: item.name, personaRecord: personaRecord)
+        } else {
+            renameNameAndRestorePersona(from: item)
+        }
+    }
+    
+    private func currentPersonaRecord(for item: PersonaImportItem) -> PersonaRecord? {
+        switch item.type {
+        case .privateKey(let privateKey):
+            let decodedData = Data(base64URLEncoded: privateKey)
+
+            let unpackedData = try? decodedData?.unpack() as? [String: Any]
+            guard let d = unpackedData?["d"] as? String else {
+                log.debug("private key error", source: "persona")
+                return nil
+            }
+            let personaRecord = PersonaRepository.queryPersonas(identifiers: nil).filter {
+                guard let privateKey = $0.privateKey else { return false }
```

Comment:
May be we can extract the filter to a function for persona, like a function named `dKeyMatch(dKey d: String) -> Bool`, and refactor the code to `PersonaRepository.queryPersonas(identifiers: nil).filter { $0.dKeyMatch(dKey: d) }`.